### PR TITLE
Lab2

### PR DESCRIPTION
**Pull request by 912-1-Cenan-Horea**

### CODE REVIEW

File: Lab2b/lexic.txt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+Identifiers:
+<letter> ::= a|b|...|z|A|B|..|Z|_
+<digit> ::=0|1|2|...|9
+<nonzero_digit> = 1|2|..|9
+<identifier> ::= <letter> | <letter><char_seq>
+<char_seq> ::= <letter><char_seq>|<digit><char_seq>
+Integer-rule:
+<digit_seq> ::= <digit> | <digit><digit_seq>
+<no> ::= <digit> | <nonzero_digit><digit_seq>
+<number> ::= 0 | <no> | -<no> | +<no>
```
Comment (916-Preda-Andrei) [inline Lab2b/lexic.txt]: "no" can be equal to 0.

File: Lab2b/lexic.txt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+character:
+<char_literal> ::= '<letter>|<digit>'
+string:
+<char> ::= <letter>|<digit>
+<string> ::= <char> | <char><string>
+<constchar> ::= "string"
```
Comment (916-Preda-Andrei) [inline Lab2b/lexic.txt]: You forgot '<' and '>'. You should put the string between these 2

File: Lab2b/lexic.txt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+<digit> ::=0|1|2|...|9
+<nonzero_digit> = 1|2|..|9
+<identifier> ::= <letter> | <letter><char_seq>
+<char_seq> ::= <letter><char_seq>|<digit><char_seq>
+Integer-rule:
+<digit_seq> ::= <digit> | <digit><digit_seq>
+<no> ::= <digit> | <nonzero_digit><digit_seq>
+<number> ::= 0 | <no> | -<no> | +<no>
+character:
+<char_literal> ::= '<letter>|<digit>'
```
Comment (916-Preda-Andrei) [inline Lab2b/lexic.txt]: Wouldn't be better to write it this way `'<letter>' | '<digit>'`?

File: Lab2b/syntax.txt [hunk 4]
```diff
@@ -0,0 +1,22 @@
+Instructions are separated by ";"
+<type> ::= int|char|string
+<decl> ::= <type> <identifier>;
+<arraydecl> ::= <type> <identifier>[<no>];
```
Comment (916-Preda-Andrei) [inline Lab2b/syntax.txt:4]: Is it ok to declare an array with size 0?

File: Lab2b/syntax.txt [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+<array_identifier> ::= <identifier>[<no>]|<identifier>[<identifier>]
+<declstmt> ::= <decl> | <arraydecl>
+<prgrm> ::= <cmpdstmt>
+<cmpdstmt> ::= {<stmtlist>}
+<stmtlist> ::= <stmt>|<stmt>;<stmtlist>
+<stmt> ::= <simplestnt>|<structstmt>
```
Comment (916-Preda-Andrei) [inline Lab2b/syntax.txt]: `<simplestnt>` does not exist. You declared an `<simplstmt>` instead.

File: Lab2b/syntax.txt [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+<simplstmt> ::= <assignstmt>|<iostmt>|<declstmt> 
+<assignstmt> ::= <identifier>=<expression>;
+<expression> ::= <expression>+<term>|<expression>-<term>|<term>
+<term> ::= <term>*<factor>|<term>/<factor>|<factor>
+<factor> ::= (<expression>)|<idetntifier>|<array_identifier>
+<iostmt> ::= read(<identifier>|<array_identifier>);|write(<identifier>|<array_identifier>);
+<structstmt> ::= <cmpdstmt> | <ifstmt> | <whilestmt>
+<ifstmt> ::= if(<condition>)<cmpdstmt> | <ifstmt>else <cmpdstmt> | <ifstmt>else <ifstmt>
+<forstmt> ::= for(<declstmt>;<condition>;<simplestmt>)<cmpdstmt>
```
Comment (916-Preda-Andrei) [inline Lab2b/syntax.txt:19]: Where do you use it? You can add it to the `<structstmt>`

File: Lab2b/tokens.txt [hunk 7]
```diff
@@ -0,0 +1,30 @@
++
+-
+*
+/
+%
```
Comment (916-Preda-Andrei) [inline Lab2b/tokens.txt:5]: Can't find this in the sintax file

File: Lab2b/syntax.txt [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+<cmpdstmt> ::= {<stmtlist>}
+<stmtlist> ::= <stmt>|<stmt>;<stmtlist>
+<stmt> ::= <simplestnt>|<structstmt>
+<simplstmt> ::= <assignstmt>|<iostmt>|<declstmt> 
+<assignstmt> ::= <identifier>=<expression>;
+<expression> ::= <expression>+<term>|<expression>-<term>|<term>
+<term> ::= <term>*<factor>|<term>/<factor>|<factor>
+<factor> ::= (<expression>)|<idetntifier>|<array_identifier>
+<iostmt> ::= read(<identifier>|<array_identifier>);|write(<identifier>|<array_identifier>);
+<structstmt> ::= <cmpdstmt> | <ifstmt> | <whilestmt>
```
Comment (916-Preda-Andrei) [inline Lab2b/syntax.txt]: No `<whilestmt>`

**Review by 916-Preda-Andrei — CHANGES_REQUESTED** [review]

File: Lab2b/syntax.txt (hunk 4 above)
Comment (912-1-Cenan-Horea) [reply Lab2b/syntax.txt:4]: Yes, from a syntactic point of view

**Pull request merged**